Makefile

```diff
@@ -0,0 +1,11 @@
+dockerimage: docker/requirements.txt
+	cd docker/ && docker build .
+
+# Exclude the editable install of this package, because:
+#  - the editable install is only useful for running tests,
+#  - we want to install Python dependencies BEFORE copying over the code
+#  (our code changes more often than dependencies)
+docker/requirements.txt:
+	pipenv lock -r | grep -v -- '-e' > $@
```

Comment:
This code also works, with the help of [requirementslib](https://github.com/sarugaku/requirementslib)
just saying
```python
from requirementslib import Lockfile, Requirement
from pathlib import Path

def dump_non_editable_deps_as_requirements_txt() -> bool:
    # argument to Lockfile.create should be the dir that hosts Pipfile*
    requirements = [Requirement.from_pipfile(package_name, config) for package_name, config in Lockfile.create('.').get_deps().items()]
    with open('docker/requirements.txt', 'w') as f:
        f.writelines([r.as_line(include_hashes=False) + '\n' for r in requirements if not r.editable])
```

Reply:
I'm not sure I understand why I'd use this approach rather than a Makefile?

Reply:
Not saying this is superior. But I love the aspect of encapsulating editable packages in an `r.editable` attribute rather than depending on string parsing `grep -v -- '-e'`. Actually your script is perfectly fine!

Reply:
Good point. String parsing is fragile, and can easily backfire! 😅